operator/v1/types_insights.go

```diff
@@ -0,0 +1,89 @@
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+//
+// Insights holds cluster-wide information about Insights.
+//
+// Compatibility level 1: Stable within a major release for a minimum of 12 months or 3 minor releases (whichever is longer).
+// +openshift:compatibility-gen:level=1
+type Insights struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata"`
+
+	// spec is the specification of the desired behavior of the Insights
+	// +kubebuilder:validation:Required
+	// +required
+	Spec InsightsSpec `json:"spec"`
+
+	// status is the most recently observed status of the Insights operator
+	Status InsightsStatus `json:"status"`
+}
+
+type InsightsSpec struct {
+	OperatorSpec `json:",inline"`
+}
+
+type InsightsStatus struct {
+	OperatorStatus `json:",inline"`
+	// gatherStatus provides basic information about the last Insights data gathering.
+	// When omitted, this means no data gathering has taken place yet.
+	GatherStatus *GatherStatus `json:"gatherStatus,omitempty"`
+	// reportStatus provides general Insights analysis results.
+	ReportStatus ReportStatus `json:"reportStatus"`
+}
+
+type GatherStatus struct {
+	// lastGatherTime is the last time when Insights data gathering finished.
+	LastGatherTime metav1.Time `json:"lastGatherTime,omitempty"`
+	// startGatherTime is the time when data gathering started. The value is 0
+	// when there is no data gathering in progress.
+	StartGatherTime metav1.Time `json:"startGatherTime,omitempty"`
+	// gatherers is a list of active gatherers (and their statuses) in the last gathering.
+	Gatherers []GathererStatus `json:"gatherers,omitempty"`
+}
+
+type ReportStatus struct {
+	// healthChecks provides basic information about active Insights health checks
+	// in a cluster
+	HealthChecks []HealthCheck `json:"healthChecks"`
+}
+
+type HealthCheck struct {
+	// id is the identificator of the respective healthcheck
```

Comment:
Is there a format or any sort of additional context we can add here? How does this relate to what the user might see in the console display?

Reply:
The example is `ccx_rules_ocp.external.rules.openshift_sdn_egress_ip_in_no_hostsubnet|OPENSHIFT_SDN_EGRESS_IP_IN_NO_HOSTSUBNET`. AFAIK user doesn't see anything from this string. It's just part of the link (maybe we could omit it).

Reply:
If it doesn't add user value immediately, I would omit it until you can think of a good use case for adding it